16.0 mig commown factor out shipping controllers

Honeyxilia opened the pull request:
(Depends on https://github.com/commown/commown-odoo-addons/pull/459 )

Honeyxilia committed:
Moved out controller and view related code to commown_shipping

Since commown_shipping already depends on portal, it's best to include the address details validation view and controller
in this module, since it's already related to commown_shipping address length limit.

Honeyxilia committed:
Removed state_id field addition to optionnal fields

The field stade_id is already present in the OPTIONAL_BILLING_FIELDS list in the portal.CustomerPortal controller.
As such, we can remove its addition.

Honeyxilia committed:
[test] Added test for details_form_validation overload

Honeyxilia committed:
Created website_sale_commown_shipping for website_sale integration of commown_shipping

Honeyxilia committed:
Removed label-optional removal for zip label

While in the website_sale.address template, the zip field label possesses the label-optional class by default,
there's a JS widget module (website_sale/static/src/js/website_sale.js:L438) which contains a method applying
the required/optional label depending on the res.country record configuration.

Since the France record ("base.fr") has zip_required by default, then the label-optional will be removed when
France is selected in the address form.

Honeyxilia committed:
Removed fields added in controller in field_required input

In the website_sale controller, the zip, city and country_id fields are already added
through the _get_mandatory_fields_billing/shipping methods. As such, they are useless here.

Honeyxilia committed:
Adapted to required shipping fields prototype change in controller

Honeyxilia committed:
Added website_sale_partner_firstname dependency for firstname/lastname fields

Honeyxilia committed:
[test] Added tests for commown_shipping validation on website_sale.address form

Honeyxilia committed:
Moved out code from commown related to website_sale_commown_shipping

fcayre commented on commown_shipping/controllers/details.py:
```diff
@@ -0,0 +1,18 @@
+
+
+class CommownShippingCustomerPortal(CustomerPortal):
+    OPTIONAL_BILLING_FIELDS = CustomerPortal.OPTIONAL_BILLING_FIELDS + [
```
On ne peut pas avoir 2 modules qui font ça donc je pense qu'il faut faire autrement : écrire la ligne suivante en haut du module ?
```
CustomerPortal.OPTIONAL_BILLING_FIELDS.extend(["street2", "state_id"])
```

Honeyxilia replied:
C'est exact - je l'avais par ailleurs remarqué en portant `commown` car j'avais fait un commit qui changeait toutes ces modifications de ce tableau en addition plutôt qu'en écriture.
Je vais donc faire ce commit dans cette PR plutôt qu'attendre de le modifier dans commown

Honeyxilia replied:
J'ai inséré ce commit dans cette PR : https://github.com/commown/commown-odoo-addons/pull/460/commits/9a3ec3e21f8957637bcade291594fb7b522d42ef

fcayre commented on commown_shipping/tests/test_controllers.py:
```diff
@@ -0,0 +1,82 @@
+        self.assertEqual(len(page.forms), 1)
+        return dict(page.forms[0].form_values())
+
+    def portal_client(self):
```
Ouch, ces méthodes commencent à être implémentées partout je pense que pour la maintenance il va falloir sortir un module dédiés aux tests du portail, qu'en penses-tu ? Pas forcément à faire dans cette PR, mais alors il faut mettre un ticket assez urgent sinon les implémentations vont dériver rapidement.

Honeyxilia replied:
Je suis d'accord - je pense qu'on pourrait même simplifier ces tests afin d'utiliser les méthodes intégrées à `HttpCase` qui demande moins de mise en place de notre côté (avec Werkzerg) pour faire les requêtes POST/GET

fcayre commented on website_sale_commown_shipping/README.rst:
```diff
@@ -67,6 +67,6 @@ Current maintainers:
-This module is part of the `commown/commown-odoo-addons <https://github.com/commown/commown-odoo-addons/tree/16.0/commown_shipping_website_sale>`_ project on GitHub.
+This module is part of the `commown/commown-odoo-addons <https://github.com/commown/commown-odoo-addons/tree/16.0/website_sale_commown_shipping>`_ project on GitHub.
 
 You are welcome to contribute.
```
Amender le commit précédent avec toutes les modifs sur README.rst et static/description/index.html

Honeyxilia replied:
Appliqué - ici, j'avais en premier temps nommé le module `commown_shipping_website_sale`, mais je pensais que la nomenclature inversée était plus logique par rapport aux autres modules c:

fcayre reviewed (changes requested):
Pas de grosses demandes ça va pouvoir rentrer vite aussi, super !

Honeyxilia committed:
Modified billing fields array override to overload

When overriding the same field array in different child classes from a common parent class (CustomerPortal),
some overrides can be practically undone by later overrides.

As such, it's preferable to directly modify the parent field array so that the changes made to the field are not overriden.

Honeyxilia commented:
/ocabot merge patch
